app/src/main/java/com/github/houseorganizer/houseorganizer/RegisterEmail.java

```diff
@@ -0,0 +1,100 @@
+package com.github.houseorganizer.houseorganizer;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.auth.FirebaseAuth;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RegisterEmail extends AppCompatActivity {
+
+    private FirebaseAuth mAuth;
+    private final String TAG = "RegisterEmail";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register_email);
+
+        mAuth = FirebaseAuth.getInstance();
+
+        findViewById(R.id.reg_email_register_button).setOnClickListener(
+                v -> {
+                    if (isValidEmail() && isValidPassword()) signUpWithEmail(v);
```

Comment:
Shouldn't we check if the email has been previously used to create another account on the app?